multiversx/mx-api-service
# Support for new nft types

## Reasoning
- In the upcoming protocol release, we have additional NFT collection types related to dynamic NFTs and NFT v2, which need accomodation at api level
  
## Proposed Changes
- In order to make the transition as smooth as possible and avoid breaking all dapps that perform queries on the collections and nfts endpoints, the new types are mapped on the existing types `NonFungibleESDT`, `SemiFungibleESDT` and `MetaESDT`, with a `subType` attribute differentiating the new types

## How to test (testnet)
- `/collections` should return also collections of subType `NonFungibleESDTv2`
- `/collections?type=NonFungibleESDT` should return also collections of subType `NonFungibleESDTv2`
- `/nfts` should return also NFTs of subType `NonFungibleESDTv2`
- `/nfts?type=NonFungibleESDT` should return also NFTs of subType `NonFungibleESDTv2`
- `/nfts/TEST-41f2aa-01` should have type `NonFungibleESDT` and subType `NonFungibleESDTv2`
- `/accounts/erd1e7qgngs0sq2f9fkyqa0ddvy52a66t6yrv9jd09afrn9cqm0lc4cquhm7ug/nfts` should return also NFTs of subType `NonFungibleESDTv2`
- `/accounts/erd1e7qgngs0sq2f9fkyqa0ddvy52a66t6yrv9jd09afrn9cqm0lc4cquhm7ug/nfts/TEST-41f2aa-01` should return the NFT of type `NonFungibleESDTv2`
